url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/rule/Consequent.java

```diff
@@ -0,0 +1,182 @@
+/*   Copyright 2013 Juan Rada-Vilela
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.fuzzylite.rule;
+
+import com.fuzzylite.Engine;
+import com.fuzzylite.FuzzyLite;
+import com.fuzzylite.factory.FactoryManager;
+import com.fuzzylite.factory.HedgeFactory;
+import com.fuzzylite.hedge.Hedge;
+import com.fuzzylite.norm.TNorm;
+import com.fuzzylite.term.Thresholded;
+import com.fuzzylite.variable.OutputVariable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.StringTokenizer;
+
+/**
+ *
+ * @author jcrada
+ */
+public class Consequent {
+
+    protected List<Proposition> conclusions;
+
+    public Consequent() {
+        this.conclusions = new ArrayList<>();
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        for (Iterator<Proposition> it = this.conclusions.iterator();
+                it.hasNext();) {
+            sb.append(it.next().toString());
+            if (it.hasNext()) {
+                sb.append(String.format(" %s ", Rule.FL_AND));
+            }
+        }
+        return sb.toString();
+    }
+
+    public void modify(double activationDegree, TNorm activation) {
+        for (Proposition proposition : conclusions) {
+            if (proposition.variable.isEnabled()) {
+                if (!proposition.hedges.isEmpty()) {
+                    for (int i = proposition.hedges.size() - 1; i >= 0; --i) {
+                        Hedge hedge = proposition.hedges.get(i);
+                        activationDegree = hedge.hedge(activationDegree);
+                    }
+                }
+
+                Thresholded term = new Thresholded();
+                term.setTerm(proposition.getTerm());
+                term.setThreshold(activationDegree);
+                term.setActivation(activation);
+                OutputVariable outputVariable = (OutputVariable) proposition.getVariable();
+                outputVariable.fuzzyOutput().getTerms().add(term);
+                FuzzyLite.logger().fine(String.format("Accumulating %s", term.toString()));
+            }
+        }
+    }
+
+    public void load(String consequent, Engine engine) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.